src/test/java/com/microsoft/sqlserver/jdbc/ActivityIDTest.java

```diff
@@ -0,0 +1,171 @@
+        config.setJdbcUrl(connectionString);
+        config.setMaximumPoolSize(poolsize);
+        ExecutorService es = Executors.newFixedThreadPool(poolsize);
+        CountDownLatch latchPoolOuterThread = new CountDownLatch(1);
+        Thread t = new Thread(new Runnable() {
+            CountDownLatch latchPool = new CountDownLatch(numPooledExecution);
+            public void run() {
+                HikariDataSource ds = new HikariDataSource(config);
+                for (int i = 0; i < numPooledExecution; i++) {
+                    es.execute(new Runnable() {
+                        public void run() {
+                            try {
+                                try (Connection con = ds.getConnection(); Statement stmt = con.createStatement()) {
+                                    stmt.execute("SELECT @@VERSION AS 'SQL Server Version'");
+                                }
+                            } catch (SQLException e) {
+                                fail(e.toString());
+                            }
+                            latchPool.countDown();
+                        }
+                    });
+                }
+                try {
+                    latchPool.await();
+                } catch (InterruptedException e) {
+                    fail(e.toString());
+                } finally {
+                    if (null != ds) {
+                        es.shutdown();
+                        ds.close();
+                    }
+                }
+                latchPoolOuterThread.countDown();
+            }
+        });
+        t.run();
+        latchPoolOuterThread.await();
+        
+        try {
+            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+                stmt.execute("SELECT @@VERSION AS 'SQL Server Version'");
+            }
+        } catch (SQLException e) {
+            fail(e.toString());
+        }
+        
+        try {
+            assertEquals(0, ActivityCorrelator.getActivityIdTlsMap().size());
+        } catch (ComparisonFailure e) {
+            Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+            logger.fine("List of threads alive:");
+            for (Thread thread: threadSet) {
+                logger.fine(thread.toString());
+            }
+            logger.fine("List of entries in the ActivityID map:");
+            logger.fine(ActivityCorrelator.getActivityIdTlsMap().toString());
+            throw new Exception(e);
+        }
+    }
+    
+    @Test
+    public void testActivityIDPooledConnection() throws Exception {
+        int poolsize = 10;
+        int numPooledExecution = 200;
+
+        PooledConnection pooledCon = ((SQLServerConnectionPoolDataSource) dsPool).getPooledConnection();
+        ExecutorService es = Executors.newFixedThreadPool(poolsize);
+        try {
+            CountDownLatch latchPool = new CountDownLatch(numPooledExecution);
+            es.execute(new Runnable() {

```

Comment:
```java
es.execute(() -> {
    IntStream.range(0, numPooledExecution).forEach(i -> {
        try (Connection c = pooledCon.getConnection(); Statement s = c.createStatement();
                ResultSet rs = s.executeQuery("SELECT @@VERSION AS 'SQL Server Version'")) {
            while (rs.next()) {
                rs.getString(1);
            }
        } catch (SQLException e) {
            fail("Pooled Thread #" + i + " failed with: " + e.toString());
        }
        latchPool.countDown();
    });
});
```

Reply:
Please make similar changes to code below, fully declaring the class/constructor for `Runnable` is unnecessary if there are no additional helper methods needed.

Reply:
Good suggestion, done.